scripts/fuel_cycle_gui.py

```diff
@@ -0,0 +1,217 @@
+        self.time_combobox = ttk.Combobox(self,width=20,textvariable=self.time_unit)
+        self.time_combobox['values'] = ('seconds','hours','days','months','years')
+        self.time_combobox.current(0)
+        self.time_divisors = {'seconds':1,'minutes':scc.minute,'hours':scc.hour,'days':scc.day,'months':scc.year/12,'years':scc.year}
+        self.old_timescale = self.time_divisors[self.time_unit.get()]
+        self.time_unit.trace('w',self.change_scale)
+        self.create_plot()
+        self.update()
+        self.entries.append(self.time_combobox)
+        self.time_combobox.grid(row=row_i,column=1)
+        row_i+=1
+        self.labeldict = { label[1][0]:float(x.get()) for label, x in zip(self.matches, self.input_entries)}
+
+        run_button = tk.Button(self,text="Run",command=self.buttonClick)
+        run_button.grid(row=row_i,column=1)
+        #test = self.apply_template(instring,self.matches,vals={x[1][0]:float(x[1][1]) for x in self.matches})
+
+    def apply_template(self,vals):
+        outstring = self.top_header
+        pointer = 0
+        outstring+='initial_condition = {:f}'.format(float(self.init_storage.get()))
+        outstring+=self.bottom_header
+        for match in self.matches:
+            outstring+=self.filtstring[pointer:match[0][0]]
+            outstring+='  [{:s}]\n    type = ConstantPostprocessor\n    execute_on = \'TIMESTEP_BEGIN INITIAL LINEAR NONLINEAR\'\n'.format(match[1][0])
+            outstring+='    value = {:g}\n  []\n'.format(vals[match[1][0]])
+            pointer = match[0][1]
+        outstring+= self.filtstring[pointer:]
+        outstring+=self.footer
+        return outstring
+
+    def create_plot(self):
+        fig = Figure(figsize=(6,4), dpi=150)
+        self.ax = fig.add_subplot(111)
+        self.canvas = FigureCanvasTkAgg(fig, master=self)
+        self.canvas._tkcanvas.grid(row=len(self.checkboxes)+11,column=2,rowspan=len(self.entries)+1 - len(self.checkboxes),columnspan=2)
+        self.canvas.draw()
+
+        self.toolbar = NavigationToolbar2Tk(self.canvas,self,pack_toolbar=False)
+        self.toolbar.update()
+        self.toolbar.grid(row=len(self.entries)+3,column=2,columnspan=2)
+        self.canvas.mpl_connect("key_press_event", self.on_key_press)
+        self.lines = self.ax.plot([],[])
+        self.ax.set_xlabel('Time [{:s}]'.format(self.time_unit.get()))
+        self.ax.set_ylabel('Tritium [kg]')
+        self.ax.set_xlim(0,8.64e8/self.time_divisors[self.time_unit.get()])
+
+    def on_key_press(self,event):
+        key_press_handler(event,self.canvas,self.toolbar)
+
+    def buttonClick(self):
+        output_string = ""
+        output_string = self.apply_template(self.labeldict)
+        with open(self.tmpfile,'w') as outfile:
+            outfile.write(output_string)
+        self.proc = subprocess.Popen([os.path.join(dir_path,'..','tmap8-opt'), '-i',self.tmpfile],stdout=subprocess.PIPE,stderr=subprocess.PIPE,cwd=os.path.dirname(self.tmpfile))
```

Comment:
This assumes `tmap8-opt` exists. This should be made more flexible, such that `tmap8-devel`, `tmap-dbg`, and `tmap8-oprof` are also acceptable.